app/src/pages/income.tsx

```diff
@@ -15,23 +12,69 @@ import Dropdown from '@components/Dropdown'
 import RequiredQuestionStatement from '@components/RequiredQuestionStatement'
 import StyledLink from '@components/StyledLink'
 
-const Income: NextPage = () => {
-  const { t } = useTranslation('common')
-  const [householdSize, setHouseholdSize] = useState<keyof typeof incomeData>()
+import type { EditablePage, IncomeData } from '@src/types'
+import { isValidIncome } from '@utils/dataValidation'
+import { initialIncomeData } from '@utils/sessionData'
+
+// Dynamically load the <IncomeRow> component to prevent SSR hydration conflicts.
+const IncomeRow = dynamic(() => import('@components/IncomeRow'), {
+  ssr: false,
+})
+
+const Income: NextPage<EditablePage> = (props: EditablePage) => {
+  // Get the session from props.
+  const {
+    session,
+    setSession,
+    reviewMode = false,
+    backRoute = '',
+    forwardRoute = '',
+  } = props
+  // Initialize form as a state with blank values.
+  const [form, setForm] = useState<IncomeData>(initialIncomeData)
+  // Use useEffect() to properly load the data from session storage during react hydration.
+  useEffect(() => {
+    setForm(session.income)
+  }, [session.income])
+
+  // If the user is reviewing previously entered data, use the review button.
+  // Otherwise, use the default button.
+  const actionButtonLabel = reviewMode ? 'updateAndReturn' : 'continue'
+
+  // Set a state for whether the form requirements have been met and the
+  // form can be submitted. Otherwise, disable the submit button.
+  const [disabled, setDisabled] = useState(true)
+  // Use useEffect() to properly load the data from session storage during react hydration
+  // Since we need to use useEffect to update this state, this also handles anytime the
+  // form state is updated, so we don't need to call the same function in handleChange().
+  useEffect(() => {
+    setDisabled(!isValidIncome(form))
+  }, [form])
+
+  // Page-specific consts.
+  // Get the allowed household sizes from the json file.
   const householdSizes: string[] = Object.keys(incomeData)
-  const incomePeriods: string[] = ['annual', 'monthly', 'biweekly', 'weekly']
+  // Get the list of allowed income periods.
+  const incomePeriods: string[] = Object.keys(
+    incomeData[householdSizes[0] as keyof typeof incomeData]
+  )
```

Comment:
Notes for reviewers: This change removes the hardcoded `incomePeriods`, making this code slightly more generic for reuse by other states and making the code less brittle if someone later updates `/public/data/income.json` but forgets to update this file. However, it does mean that the order of the keys in `/public/data/income.json` matters. Hence, the re-ordering seen in #141.